sql/restoring.sql

```diff
@@ -26,7 +26,7 @@ DECLARE
 BEGIN
     SELECT current_database() INTO db;
     EXECUTE format($$ALTER DATABASE %I RESET timescaledb.restoring $$, db);
-    RESET timescaledb.restoring;
+    SET SESSION timescaledb.restoring = 'off';
     PERFORM _timescaledb_internal.restart_background_workers();
```

Comment:
This will create problems because an explicitly set variable is written by `pg_dumpall`, which causes the restore to be confused about what setting to use. What is the problem you're trying to solve?

Reply:
Ah, thank you for the review, I wasn't sure if session variables were causing this error or not. The "problem" is that 'off' is not immediately visible in the current session after calling `timescaledb_post_restore`, which can be kind of unexpected to see (as mentioned in #4267) , since I understand that the change made by `ALTER DATABASE ... RESET` will be visible in subsequent sessions and so, the previous value is read in this session, which is still 'on'. So currently, the session needs to be restarted for `post_restore` to take effect, but I understand there is no other way. 
I'll respond and close the linked issue.

Reply:
What if it was:
```
SET SESSION timescaledb.restoring TO DEFAULT;
```
Does this solve all the problems?